CMakeLists.txt

```diff
@@ -0,0 +1,165 @@
+cmake_minimum_required (VERSION 3.2)
+
+if (NOT DEFINED CMAKE_BUILD_TYPE)
+  set (CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
+endif ()
+
+project (swig)
+
+file (STRINGS configure.ac line LIMIT_COUNT 1 REGEX "AC_INIT\\(.*\\)" )
+if (line MATCHES "AC_INIT\\(\\[(.*)\\],[ \t]*\\[(.*)\\],[ \t]*\\[(.*)\\]\\)" )
+  set (SWIG_VERSION ${CMAKE_MATCH_2})
+  set (PACKAGE_BUGREPORT ${CMAKE_MATCH_3})
+else ()
+  message (SEND_ERROR "Could not parse version from configure.ac")
+endif ()
+
+set(SWIG_ROOT ${PROJECT_SOURCE_DIR})
+
+set (SWIG_LIB share/swig/${SWIG_VERSION})
```

Comment:
Some consistent whitespace throughout this file to whatever is the CMake norm would be a nice to have.